database/src/env.rs

```diff
@@ -222,6 +227,54 @@ where
     /// a table doesn't exist.
     fn open_db_rw<T: Table>(&self, tx_rw: &Rw) -> Result<impl DatabaseRw<T>, RuntimeError>;
 
+    /// TODO
+    ///
+    /// # Errors
+    /// TODO
+    fn open_tables(&self, tx_ro: &Ro) -> Result<impl Tables, RuntimeError> {
+        Ok((
+            EnvInner::open_db_ro::<BlockInfoV1s>(self, tx_ro)?,
+            EnvInner::open_db_ro::<BlockInfoV2s>(self, tx_ro)?,
+            EnvInner::open_db_ro::<BlockInfoV3s>(self, tx_ro)?,
+            EnvInner::open_db_ro::<BlockBlobs>(self, tx_ro)?,
+            EnvInner::open_db_ro::<BlockHeights>(self, tx_ro)?,
+            EnvInner::open_db_ro::<KeyImages>(self, tx_ro)?,
+            EnvInner::open_db_ro::<NumOutputs>(self, tx_ro)?,
+            EnvInner::open_db_ro::<PrunedTxBlobs>(self, tx_ro)?,
+            EnvInner::open_db_ro::<PrunableHashes>(self, tx_ro)?,
+            EnvInner::open_db_ro::<Outputs>(self, tx_ro)?,
+            EnvInner::open_db_ro::<PrunableTxBlobs>(self, tx_ro)?,
+            EnvInner::open_db_ro::<RctOutputs>(self, tx_ro)?,
+            EnvInner::open_db_ro::<TxIds>(self, tx_ro)?,
+            EnvInner::open_db_ro::<TxHeights>(self, tx_ro)?,
+            EnvInner::open_db_ro::<TxUnlockTime>(self, tx_ro)?,
+        ))
+    }
+
+    /// TODO
+    ///
+    /// # Errors
+    /// TODO
+    fn open_tables_mut(&self, tx_rw: &Rw) -> Result<impl TablesMut, RuntimeError> {
+        Ok((
+            EnvInner::open_db_rw::<BlockInfoV1s>(self, tx_rw)?,
+            EnvInner::open_db_rw::<BlockInfoV2s>(self, tx_rw)?,
+            EnvInner::open_db_rw::<BlockInfoV3s>(self, tx_rw)?,
+            EnvInner::open_db_rw::<BlockBlobs>(self, tx_rw)?,
+            EnvInner::open_db_rw::<BlockHeights>(self, tx_rw)?,
+            EnvInner::open_db_rw::<KeyImages>(self, tx_rw)?,
+            EnvInner::open_db_rw::<NumOutputs>(self, tx_rw)?,
+            EnvInner::open_db_rw::<PrunedTxBlobs>(self, tx_rw)?,
+            EnvInner::open_db_rw::<PrunableHashes>(self, tx_rw)?,
+            EnvInner::open_db_rw::<Outputs>(self, tx_rw)?,
+            EnvInner::open_db_rw::<PrunableTxBlobs>(self, tx_rw)?,
+            EnvInner::open_db_rw::<RctOutputs>(self, tx_rw)?,
+            EnvInner::open_db_rw::<TxIds>(self, tx_rw)?,
+            EnvInner::open_db_rw::<TxHeights>(self, tx_rw)?,
+            EnvInner::open_db_rw::<TxUnlockTime>(self, tx_rw)?,
+        ))
+    }
```

Comment:
2. `trait EnvInner` gives access to this, but instead of returning the messy tuple, it returns a nice clean `impl TablesMut` object - this is also more type safe as the tuple is bounded by the blanket impl from before - these functions are also pre-implemented since they aren't backend specific